src/gsy_e/models/strategy/energy_parameters/heatpump/heat_pump.py

```diff
@@ -206,8 +221,13 @@ def update_tanks_temperature(
         traded_heat_energy_kJ = self.calc_Q_kJ_from_energy_kWh(
             last_time_slot, bought_energy_kWh, source_temp_C
         )
-        self.charger.charge(traded_heat_energy_kJ, last_time_slot)
-        self.charger.update_tanks_temperature(time_slot)
+        net_energy_kJ = traded_heat_energy_kJ - heat_demand_kJ
```

Comment:
Ok , now I see where the differentiation between charge / discharge happens. Makes sense